src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
-    @Override
-    public void add(T value) {
+    private static final int DEFAULT_CAPACITY = 10;
+    private int size;
+    private Object[] elementData;
 
+    public ArrayList() {
+        this.elementData = new Object[DEFAULT_CAPACITY];
     }
 
     @Override
-    public void add(T value, int index) {
+    public boolean add(T value) {
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        elementData[size] = value;
+        size++;
+        return true;
+    }
 
+    @Override
+    public void add(T value, int index) {
+        rangeCheckForAdd(index);
+        if (size == elementData.length) {
+            elementData = grow();
+        }
+        System.arraycopy(elementData, index,
+                elementData, index + 1,
+                size - index);
+        elementData[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (size + list.size() > elementData.length) {
+            elementData = grow();
+        }
+        for (int i = 0; i < list.size(); i++) {
+            elementData[size++] = list.get(i);
+        }
+    }
 
+    private void checkIndex(int index) {
+        if ((index >= size || index < 0)) {
```

Comment:
```suggestion
    private void checkIndex(int index) {
        if (index >= size || index < 0) {
```
Private ones (methods, classes) should be after public ones.